client/landing/stepper/declarative-flow/site-setup-flow.ts

```diff
@@ -103,24 +104,146 @@ export const siteSetupFlow: Flow = {
+				break;
+		}
+
+		switch ( intent ) {
+			case SiteIntent.Write: {
+				switch ( currentStep ) {
+					case 'options':
+						updateFlowProgress( { middleProgress: { progress: 1, count: 4 } } );
+						break;
+					case 'bloggerStartingPoint':
+						updateFlowProgress( { middleProgress: { progress: 2, count: 4 } } );
+						break;
+					case 'courses':
+					case 'designSetup':
+						updateFlowProgress( { middleProgress: { progress: 3, count: 4 } } );
+						break;
+					case 'processing':
+						updateFlowProgress( { endStep: currentStep } );
+						break;
+				}
+
+				break;
+			}
+			case SiteIntent.Build: {
+				switch ( currentStep ) {
+					case 'designSetup':
+						updateFlowProgress( { middleProgress: { progress: 1, count: 2 } } );
+						break;
+					case 'processing':
+						updateFlowProgress( { endStep: currentStep } );
+						break;
+				}
+
+				break;
+			}
+			case SiteIntent.Sell: {
+				switch ( currentStep ) {
+					case 'options':
+						updateFlowProgress( { middleProgress: { progress: 1, count: 6 } } );
+						break;
+					case 'storeFeatures':
+						updateFlowProgress( { middleProgress: { progress: 2, count: 6 } } );
+						break;
+				}
+				if ( storeType === 'simple' ) {
+					switch ( currentStep ) {
+						case 'designSetup':
+							updateFlowProgress( { middleProgress: { progress: 3, count: 6 } } );
+							break;
+						case 'processing':
+							updateFlowProgress( { endStep: currentStep } );
+							break;
+					}
+				} else if ( storeType === 'power' ) {
+					switch ( currentStep ) {
+						case 'storeAddress':
+							updateFlowProgress( { middleProgress: { progress: 3, count: 6 } } );
+							break;
+						case 'businessInfo':
+							updateFlowProgress( { middleProgress: { progress: 4, count: 6 } } );
+							break;
+						case 'wooConfirm':
+						case 'wooInstallPlugins':
+							updateFlowProgress( { middleProgress: { progress: 5, count: 6 } } );
+							break;
+						case 'processing':
+							updateFlowProgress( { endStep: currentStep } );
+							break;
+					}
+				}
+
+				break;
+			}
+			case SiteIntent.Import: {
+				switch ( currentStep ) {
+					case 'import':
+						updateFlowProgress( { middleProgress: { progress: 1, count: 4 } } );
+						break;
+					case 'importList':
+					case 'importReady':
+					case 'importReadyNot':
+					case 'importReadyWpcom':
+					case 'importReadyPreview': {
+						updateFlowProgress( { middleProgress: { progress: 2, count: 4 } } );
+						break;
+					}
+					case 'importerWix':
+					case 'importerBlogger':
+					case 'importerMedium':
+					case 'importerSquarespace':
+					case 'importerWordpress':
+						updateFlowProgress( { middleProgress: { progress: 3, count: 4 } } );
+						break;
+					case 'processing':
+						updateFlowProgress( { endStep: currentStep } );
+						break;
+				}
+
+				break;
 			}
-		} else {
-			setStepProgress( undefined );
+		}
+
+		function updateFlowProgress( flowProgress: {
```

Comment:
My two cents in this way:

I think we only need a `currentStep` and a `progress`.
* For the `currentStep`, we can resolve the current stage (begin, middle, end) inside this function as you also define `beginningSteps`  here.
* For the `progress`, people can decide how to increase the process by this value.

What do you think?

Reply:
Thanks for suggestion, I'll look into it